internal/router/router.go

```diff
@@ -7,60 +7,29 @@ import (
 	"github.com/ethereum/go-ethereum/rpc"
 	"github.com/gorilla/mux"
 	"github.com/rs/zerolog/log"
-	"github.com/textileio/go-tableland/internal/chains"
 	"github.com/textileio/go-tableland/internal/router/controllers"
 	"github.com/textileio/go-tableland/internal/router/middlewares"
 	"github.com/textileio/go-tableland/internal/router/rpcservice"
-	systemimpl "github.com/textileio/go-tableland/internal/system/impl"
+	"github.com/textileio/go-tableland/internal/system"
 	"github.com/textileio/go-tableland/internal/tableland"
-	"github.com/textileio/go-tableland/internal/tableland/impl"
-	"github.com/textileio/go-tableland/pkg/parsing"
-	"github.com/textileio/go-tableland/pkg/sqlstore"
-	sqlstoreimpl "github.com/textileio/go-tableland/pkg/sqlstore/impl"
-	"github.com/textileio/go-tableland/pkg/sqlstore/impl/user"
 )
 
 // ConfiguredRouter returns a fully configured Router that can be used as an http handler.
 func ConfiguredRouter(
-	extURLPrefix string,
-	metadataRendererURI string,
+	tableland tableland.Tableland,
+	systemService system.SystemService,
 	maxRPI uint64,
 	rateLimInterval time.Duration,
-	parser parsing.SQLValidator,
-	userStore *user.UserStore,
-	chainStacks map[tableland.ChainID]chains.ChainStack,
```

Comment:
Main change in this file is to make the `ConfiguredRouter` function accept instances of interfaces instead of configuration data to create those instances internally. This is more consistent with good dependency injection design and makes writing tests that use this function better since you can pass in mocks here.